web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,203 @@
+from urllib.error import HTTPError
+from xml.dom import NotFoundErr
+
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from fake_useragent import UserAgent
+
+BASE_URL = "https://ww1.gogoanime2.org"
+
+
+def search_scraper(anime_name: str) -> list:
+
+    """[summary]
+
+    This function will take an url and
+    return list of anime after scraping the site.
+
+    >>> type(search_scraper("demon_slayer"))
+    <class 'list'>
+
+    Args:
+        anime_name (str): [Name of anime]
+
+    Raises:
+        e: [Raises exception on failure]
+
+    Returns:
+        [list]: [List of animes]
+    """
+
+    try:
```

Comment:
try / except block should be just around critical areas of code.  They should not be 30 lines long!!

Reply:
removed the long try catches with critical ones